### DISCUSSION TITLE
GitLens is not showing annotations on files, why?

### DISCUSSION THREAD

**Discussion by eamodio** [gitkraken/vscode-gitlens#1299] [Q&A] [answered]
There are often reports of GitLens annotations (current line, blame, etc) not showing up or not showing up anymore.

**Comment by eamodio** [comment, marked as answer]
This is typically caused by a known "issue" with VS Code & Git on file systems that aren't case sensitive (Windows, macOS). It stems from VS Code not being able to get the correct casing of the folder/file on disk, so it uses whatever casing you provide. And Git (unless you set the `core.ignoreCase` config setting to `true`) is expecting the folder/file casings to match exactly as they are on disk.

To "fix" this either open the folder/workspace with the exact casing as it exists on disk, or change your Git config to set `core.ignoreCase` to `true`, by running `git config --global core.ignoreCase true` on the command line.

**Reply by ferenczy** [reply]
I'll rather configure Git to ignore the casing, that seems to be safer to me, I'm just curious, what exactly does it mean "_open the folder/workspace with the exact casing as it exists on disk_"?

I have opened VS Code simply by executing `code .` in the solution's root directory and got the following notification:

> This workspace was opened with a different casing than what exists on disk. Please re-open this workspace with the exact casing as it exists on disk, otherwise you may experience issues with certain Git features, such as missing blame or history.

**Reply by trey-miller** [reply]
Hi, I don't know if things changed or if my situation is different, but this doesn't seem to work for me in VSCode on MacOS. Basically I've got a file that was renamed in a git commit by only changing the case, but GitLens File History does not understand this. 

I have set the global git config `core.ignoreCase` set to `true` as you suggest above. I have also tried toggling the "File History Follows Renames". Neither of these settings fix the problem. If I am in the file that was renamed, and go to GitLens Inspect > File History, the file name it shows is the old/wrong case, and there is no history. Likewise, there are no blame annotations. I've even tried uninstalling GitLens, deleting the entire folder for the extension in `~/.vscode/extensions` and reinstalling, hoping it was something stored in there. No help. Is there something else I can try, or should I file this as a new issue?

**Comment by MasonChinkin** [comment]
+1 to @ferenczy's follow up.

@eamodio what does it mean to "open with a different casing" when we are just running `code .` in the terminal?

(Really appreciate your work making this!)

**Reply by eamodio** [reply]
If you are using `code .`, I'm confused as to why the casing would be wrong. Typically, it happens if you did something like `code c:\foo` when the folder on disk is `c:\Foo`.

**Reply by eamodio** [reply]
@bpasero Do you have any idea why `code .` would cause VS Code to open the current folder with a casing different than what is on disk?

**Reply by bpasero** [reply]
I think this can happen when the terminal was explicitly instructed to open on the "wrong" casing. Same as VSCode, the terminal is not validating the case and just uses it since files can be accessed just fine.
